packages/client/lib/sync/fetcher/blockfetcher.ts

```diff
@@ -24,13 +24,13 @@ export class BlockFetcher extends BlockFetcherBase<Block[], Block> {
   async request(job: Job<JobTask, Block[], Block>): Promise<Block[]> {
     const { task, peer } = job
     const { first, count } = task
-    const headers = await (peer!.eth as EthProtocolMethods).getBlockHeaders({
-      block: first,
-      max: count,
-    })
-    const bodies: BlockBodyBuffer[] = <BlockBodyBuffer[]>(
-      await peer!.eth!.getBlockBodies(headers.map((h) => h.hash()))
-    )
+    const headers = (
+      await (peer!.eth as EthProtocolMethods).getBlockHeaders({
+        block: first,
+        max: count,
+      })
+    )[1]
+    const bodies = (await peer!.eth!.getBlockBodies({ hashes: headers.map((h) => h.hash()) }))[1]
```

Comment:
thanks for the in-depth test runs, yeah this is tricky and at first I thought it was because of [this change](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1324/files#diff-5cdab2534be43ab3aaced39ef3b1124d0329d2c8c897b9343ff914c02b227b36L187-R187) but that turns out to be in a separate PR 😄 thanks for the code suggestion that seems reasonable I'll give it a run today and if it handles the issue well I will merge it in :)

Reply:
If I'm reading the logs correctly, it looks like the peer just isn't returning anything, so maybe they're missing a block for some reason?  It wasn't obvious to me from the logs but it didn't seem to break anything.